### PR TITLE
feat: [PL-26858]: Centralised view for User Details

### PR DESCRIPTION
**Pull request by reetika04**
#### Screenshots
Screenshot 2022-08-18 at 3 45 57 PM
Screenshot 2022-08-18 at 3 55 11 PM

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by diptiman-adak** [comment]
Quality gate Quality gate: **_Failed_** 
 


            
 #### 99 Issues 
* Code issue 99 Code issues 

 [Details](https://app.embold.io/organization/gh/harness/repositories/e864badd0b95ab81c060b15b85f15701/pull-request/10636)

File: src/modules/20-rbac/strings/strings.en.yaml [hunk 1]
```diff
@@ -134,12 +135,16 @@ usersPage:
   unlockSuccessMessage: User '{{name}}' unlocked successfully
   lockedOutLabel: (Locked out)
 userDetails:
+  invalidScopeText: 'You are in the scope: {{scope}}. Please select the organization for which you want to view results. or'
+  scopeAll: 'Switch to Scope: All'
+  groupMemberships: Group Memberships
   noMembersMessage: There are no members added to this User Group
+  noUserGroups: There are no user groups found for this user. Click on the button below to start the user to a group
```
Comment (upadhyay-an) [inline src/modules/20-rbac/strings/strings.en.yaml:142]: `Click on the button below to **add** the user to a group

File: src/modules/20-rbac/strings/strings.en.yaml [hunk 2]
```diff
@@ -134,12 +135,16 @@ usersPage:
   unlockSuccessMessage: User '{{name}}' unlocked successfully
   lockedOutLabel: (Locked out)
 userDetails:
+  invalidScopeText: 'You are in the scope: {{scope}}. Please select the organization for which you want to view results. or'
```
Comment (upadhyay-an) [inline src/modules/20-rbac/strings/strings.en.yaml:138]: I think the `or` at the end of sentence is unintended?

File: src/modules/20-rbac/pages/UserDetails/views/UserGroupTable.tsx [hunk 3]
```diff
@@ -28,46 +28,54 @@ import type { PipelineType, ProjectPathProps, UserPathProps } from '@common/inte
...
+    data.users?.map(user => {
+      return { email: user.email, name: user.name }
+    }) || []
+
+  return <AvatarGroup avatars={avatars} restrictLengthTo={6} />
 }
 
 const ResourceGroupColumnMenu: Renderer<CellProps<UserGroupDTO>> = ({ row, column }) => {
   const data = row.original
+  const { accountIdentifier = '', orgIdentifier, projectIdentifier, identifier } = data
```
Comment (upadhyay-an) [inline src/modules/20-rbac/pages/UserDetails/views/UserGroupTable.tsx:68]: ```const { accountIdentifier = '', orgIdentifier, projectIdentifier, identifier = '' } = data```
Giving a fallback value for identfier shall keep it consistent with what we had on line 68 earlier.

File: src/modules/20-rbac/strings/strings.en.yaml (hunk 2 above)
Comment (reetika04) [reply src/modules/20-rbac/strings/strings.en.yaml:139]: Yes

**Review by mayankVermaHarness — DISMISSED** [review]

**Comment by reetika04** [comment]
retrigger eslint

**Comment by reetika04** [comment]
retrigger jest

**Review by mayankVermaHarness — APPROVED** [review]

**Review by upadhyay-an — APPROVED** [review]

**Review by abhinavrastogi-harness — APPROVED** [review]

**Pull request merged**